src/litdata/streaming/downloader.py

```diff
@@ -233,6 +234,82 @@ def download_file(self, remote_filepath: str, local_filepath: str, remote_chunk_
                     os.remove(temp_path)
 
 
+class DBFSDownloader(Downloader):
+    def __init__(
+        self,
+        remote_dir: str,
+        cache_dir: str,
+        chunks: List[Dict[str, Any]],
+        storage_options: Optional[Dict] = {},
+    ):
+        if not _DATABRICKS_SDK_AVAILABLE:
+            raise ModuleNotFoundError(
+                "Support for Downloading dbfs dataset depends on `databricks-sdk`.",
+                "Please, run: `pip install databricks-sdk",
+            )
+
+        super().__init__(
+            remote_dir=remote_dir,
+            cache_dir=cache_dir,
+            chunks=chunks,
+            storage_options=storage_options,
+        )
+
+        from databricks.sdk import WorkspaceClient
+
+        self._dbfs_client: Optional[WorkspaceClient] = None
+
+    def download_file(self, remote_filepath: str, local_filepath: str, remote_chunk_filename: str = "") -> None:
+        from databricks.sdk.core import DatabricksError
+
+        if self._dbfs_client is None:
+            self._create_dbfs_client()
+        assert self._dbfs_client is not None
+
+        obj = parse.urlparse(remote_filepath)
+        if obj.scheme != "dbfs":
+            raise ValueError(f"Expected scheme to be `dbfs`, instead, got {obj.scheme} for remote={remote_filepath}")
+
+        if os.path.exists(local_filepath):
+            return
+
+        with suppress(Timeout), FileLock(
+            local_filepath + ".lock", timeout=3 if obj.path.endswith(_INDEX_FILENAME) else 0
+        ):
+            local_tmp = local_filepath + ".tmp"
+            response = self._dbfs_client.files.download(obj.path).contents
+
+            assert response is not None
+
+            try:
+                with response, open(local_tmp, "wb") as f:
+                    for chunk in iter(lambda: response.read(1024 * 1024), b""):
+                        f.write(chunk)
+
+                os.rename(local_tmp, local_filepath)
+            except DatabricksError as e:
+                if e.error_code == "PERMISSION_DENIED":
+                    e.args = (
+                        "Ensure the file path or credentials are set correctly. For "
+                        + "Databricks Unity Catalog, file path must starts with `dbfs:/Volumes` "
+                        + "and for Databricks File System, file path must starts with `dbfs`. "
+                        + e.args[0],
+                    )
+                raise e
+            except Exception as e:
+                raise e
+            finally:
+                # Ensure cleanup of temp file if an error occurs
+                if os.path.exists(local_tmp):
+                    os.remove(local_tmp)
+
+    def _create_dbfs_client(self) -> None:
+        """Create a Databricks File System client."""
+        from databricks.sdk import WorkspaceClient
+
+        self._dbfs_client = WorkspaceClient()
```

Comment:
I think we could define the client above the `download_file` function, similar to how it's done in other downloaders.  
Additionally, we will need to pass the `storage_options`.
>   storage_options is used in case user wants to provide the credentials directly through StreamingDataset 

Based on the given [documentation](https://docs.databricks.com/aws/en/dev-tools/sdk-python#authenticate-the-databricks-sdk-for-python-with-your-databricks-account-or-workspace), it seems we have two parameters: `host` and `token`.  
So, we could include them(storage options fields) in the README documentation as well for reference.

Reply:
Also, I see that a few parts are taken from Mosaic.  
@tchaton, should we include a reference to Mosaic here?  

https://github.com/mosaicml/streaming/blob/89d7ed1e8bbcaff1bc8ac7c11464002bdc3b655a/streaming/base/storage/download.py#L728